projects/ideabox-groups.md

```diff
@@ -0,0 +1,210 @@
+---
+title: IdeaBox Groups
+---
+
+Every developer has more ideas than time. As David Allen likes to say "the human brain is for creating
+ideas, not remembering them." In this project, we'll be building a simple application for recording and
+archiving our ideas (good and bad alike).
+
+Throughout the project, one of our focuses will be on providing a fluid and responsive client-side
+interface. To this end, we'll rely on JavaScript and jQuery to implement snappy filtering in the
+browser, and `localStorage` to persist our wonderful ideas between sessions.
+
+## Project Requirements
+
+### Architecture
+
+For this project, we'll be increasingly thinking about the "data model" and "dom model" as separate entities. We'll be using:
+
+- JSON and `localStorage` to persist data on page reload.
+- JavaScript to manage client-side interactions.
+
+Your entire application will consist of one HTML page or template.
+You will have two javascript files:
+
+1. An `idea.js` file that contains an `Idea` class. This must be the first script so that your `main.js` file has access to your `Idea` class.
+  * `Idea` methods must include, but are not limited to:
+    1. `constructor`
+    2. `saveToStorage`
+    3. `deleteFromStorage`
+    4. `updateSelf`
+    5. `updateQuality`
+2. A `main.js` file that contains all dom related javascript.
+
```

Comment:
I love this idea of dividing the javascript into 2 files and separating the logic.  Cheers to this!